sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -38,28 +50,35 @@ export class ManagedIdentityCredential implements TokenCredential {
    * @param clientId - The client ID of the user-assigned identity, or app registration (when working with AKS pod-identity).
    * @param options - Options for configuring the client which makes the access token request.
    */
-  constructor(clientId: string, options?: TokenCredentialOptions);
+  constructor(clientId: string, options?: ManagedIdentityCredentialOptions);
```

Comment:
This would be a breaking change no? Wouldn't the user get a TS error if they try to pass the less specific `TokenCredentialOptions` with this change in place? If we still need to add the `ManagedIdentityCredentialOptions` interface, it seems like these would have to take a type union to avoid breaking.

Reply:
users won’t get an error as long as the properties of ManagedIdentityCredentialOptions are only additive. Also, we’re still in v2-beta! that helps

Reply:
Unless we added a required property to `ManagedIdentityCredentialOptions` which wasn't on `TokenCredentialOptions` correct? Also, while I realize we're bumping the major version, we should still try to limit breaking changes where possible. This constructor was part of the GA API in 1.* so I think if simply having a type union here makes it compatible it's probably worth it to limit the code changes necessary to upgrade from 1.* to 2*

Reply:
Right, but we’re not adding a required property, since you mentioned that ManagedIdentityCredential shouuldn’t have tenantId on the options. So, this is going back to a TokenCredentialOptions.